pkg/apis/v1/nodeclaim.go

```diff
@@ -73,6 +73,12 @@ type NodeClaimSpec struct {
 	// +kubebuilder:validation:Schemaless
 	// +optional
 	ExpireAfter NillableDuration `json:"expireAfter,omitempty"`
+	// MinimumPriceImprovementPercent is the minimum price improvement necessary to disrupt this node, as an integer percentage.
+	// The default is 0%, which maintains the existing consolidation behavior prior to this feature.
+	// +kubebuilder:validation:Minimum:=0
+	// +kubebuilder:validation:Maximum:=100
+	// +optional
+	MinimumPriceImprovementPercent *int32 `json:"minimumPriceImprovementPercent,omitempty"`
```

Comment:
I think this one makes sense at the NodePool level, rather than the NodeClaim level. This is because the decision to consolidate is at the pool level, unlike expiry/tgp, which make sense outside of the context of a pool (i.e. standalone nodeclaim).

Reply:
I put it in NodeClaim because I think it could be helpful to have the flexibility to set different improvement thresholds for each NodeClaim in a shared NodePool. The existing candidate filtering logic already considers the price of each NodeClaim individually, so there's no added complexity there.